engine/engine.go

```diff
@@ -62,12 +64,18 @@ func New(opts Opts) v1.QueryEngine {
 	return &compatibilityEngine{
 		core: core,
 		prom: promql.NewEngine(opts.EngineOpts),
+		fallbacks: promauto.With(opts.Reg).NewCounter(
+			prometheus.CounterOpts{
+				Name: "thanos_engine_fallbacks_total",
```

Comment:
Maybe we can add a label to denote if the engine did fallback, and the metric could be `thanos_engine_queries_total` instead? It would be useful to see the percentage of queries that fallback under real-world use. 🙂

Reply:
Updated. Thanks for the suggestion!